classes/mutex/PgAdvisoryMutex.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace malkusch\lock\mutex;
+
+use malkusch\lock\exception\LockAcquireException;
+use malkusch\lock\exception\LockReleaseException;
+
+/**
+ * @author Dmitrii Korotovskii <dmitry@korotovsky.io>
+ * @license WTFPL
+ */
+class PgAdvisoryMutex extends LockMutex
+{
+    /**
+     * @var \PDO $pdo The PDO.
+     */
+    private $pdo;
+
+    /**
+     * @var array The resource map <String, Integer>
+     */
+    private $map;
+
+    /**
+     * @var string
+     */
+    private $resourceType;
+
+    /**
+     * @var int
+     */
+    private $resourceId;
+
+    /**
+     * PgAdvisoryMutex constructor.
+     * @param \PDO    $pdo           The PDO connection
+     * @param array   $map           Resource map
```

Comment:
What's this `resourceMap` and why is it passed in full instead of only the key for `$resourceType`?

Reply:
Because the function `pg_try_advisory_lock(key1 int, key2 int)` has definition either both `int` args or one `bigint`. So I could do `crc32` to make `int` out of string key, but potentially it could lead to collisions. So the end-user have to define its own map of keys and its ids for PostgreSQL.

Reply:
What about hasing the key (e.g. sha256) using raw mode and then converting part of the output to two 32 bit integers using `bindec()`.